diff --git a/hintless_simplepir/new_pir_test.cc b/hintless_simplepir/new_pir_test.cc
index 0a20957..e0fa889 100644
--- a/hintless_simplepir/new_pir_test.cc
+++ b/hintless_simplepir/new_pir_test.cc
@@ -45,8 +45,8 @@ namespace {
 
 using RlweInteger = Parameters::RlweInteger;
 
-const int rows_db = 2048;
-const int cols_db = 1024;
+const int rows_db = 32768;
+const int cols_db = 16384;
 bool use_static_db = true;
 
 const Parameters kParameters{
@@ -57,7 +57,7 @@ const Parameters kParameters{
     .offline_lwe_secret_dim = 2816,
     .lwe_modulus_bit_size = 32,
     .lwe_plaintext_bit_size = 8, // <= 8
-    .db_stack_cells = 1,
+    .db_stack_cells = 8,
     .lwe_error_variance = 8,
     .linpir_params =
         linpir::RlweParameters<RlweInteger>{
